server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1908,4 +1916,71 @@ public String getSequenceName()
     }
   }
 
+  public static class DataStoreMetadataUpdateResult
+  {
+    private final boolean failed;
+    private final boolean canRetry;
+    @Nullable private final String errorMsg;
```

Comment:
This class is a good idea, but we're jumping to error messages too soon. This class hold the exception, not the error message. The exception class conveys information. The exception has a stack trace. With the message, developers have no information about what specific bit of code went wrong when debugging. Convert the exception to a string at the point the string is needed.

Reply:
I think we should revisit this class again once your error PR is in. and consider replacing it with a Fault.